nexus/db-model/src/image.rs

```diff
@@ -3,14 +3,15 @@
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
 use super::{BlockSize, ByteCount, Digest};
-use crate::schema::image;
+use crate::schema::{image, project_image, silo_image};
 use db_macros::Resource;
 use nexus_types::external_api::views;
 use nexus_types::identity::Resource;
+use omicron_common::api::external::Error;
 use serde::{Deserialize, Serialize};
 use uuid::Uuid;
 
-// Project images
+// Shared image definition
```

Comment:
It's not that hard to see this from the `TryFrom` impls and stuff, but I think it would be helpful to add a comment to the effect of:

```
// Image, ProjectImage, and SiloImage are almost identical. Image is the model
// for the underlying table and ProjectImage and SiloImage are backed by views
// of that table. They have all the same fields except ProjectImage has both a
// silo_id and a project_id, while SiloImage only has a silo_id. Image has a
// silo_id and an optional project_id to cover both possibilities
```